pkg/xds/generator/admin_proxy_generator.go

```diff
@@ -97,8 +97,7 @@ func (g AdminProxyGenerator) Generate(ctx xds_context.Context, proxy *core_xds.P
 		if err != nil {
 			return nil, err
 		}
-		// todo(lobkovilya): handle ZoneEgress
-		if (proxy.Dataplane != nil || proxy.ZoneIngress != nil) && hasCpValidationCtx {
+		if (proxy.Dataplane != nil || proxy.ZoneIngress != nil || proxy.ZoneEgressProxy != nil) && hasCpValidationCtx {
```

Comment:
I mean... one of them is always not nil, right? shouldn't `hasCpValidationCtx` be enough?

Reply:
Right, thanks!